### PR TITLE
Add code to update all ssl cert files used in the test system

### PR DESCRIPTION
**Pull request by MSP-Greg**
### Description

Currently, three scripts update ssl cert files, and all have notes (some incomplete) about required shell commands to perform additional tasks.

This is cumbersome, and creates more work to maintain the files.

A bash script is provided that updates all ssl related files, all with a four year time limit.

Additionally, unused files and folders have been removed.

An md file at [examples/generate_all_certs.md](https://github.com/puma/puma/compare/master...MSP-Greg:00-certs-update?expand=1#diff-aa18dc9707c95b21eba8b4b4a3bf1278413b64d63d874a44f936fc4b936645d5) gives the simple instructions for running the script.

I have only run it on Ubuntu, but it should run on macOS (I think).  If anyone can try that in their fork, it would be appreciated...

### Your checklist for this pull request
- [x] I have reviewed the [guidelines for contributing](../blob/master/CONTRIBUTING.md) to this repository.
- [ ] I have added (or updated) appropriate tests if this PR fixes a bug or adds a feature.
- [ ] My pull request is 100 lines added/removed or less so that it can be easily reviewed.
- [ ] If this PR doesn't need tests (docs change), I added `[ci skip]` to the title of the PR.
- [ ] If this closes any issues, I have added "Closes `#issue`" to the PR description or my commit messages.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed, including Rubocop.

### CODE REVIEW

File: examples/generate_all_certs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+SCRIPT=$(realpath -s "$0")
+CERT_HOME=$(dirname "$SCRIPT")
+
+#——————————————————————————————— process ssl files in examples/puma
+echo -e "$Dash process ssl files in examples/puma$Reset"
+cd $CERT_HOME/puma
+ruby ../generate_server_test.rb
+printf "Done with ../generate_server_test.rb\n\n"
+
+if [ -f server.p12 ]; then rm server.p12; fi
```
Comment (tannakartikey) [inline examples/generate_all_certs:19]: `rm -f <file_name>` does not throw an error if the file does not exist.

File: examples/generate_all_certs [hunk 2]
```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
```
Comment (tannakartikey) [inline examples/generate_all_certs:1]: The file is not executable.

File: examples/generate_all_certs.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+## Updating Puma test cert files
+
+One must have openssl and keytool in their Path.  Keytool is part of the JDK.
+
+It maybe helpful to install the 'certificate_authority' gem.
```
Comment (tannakartikey) [inline examples/generate_all_certs.md:5]:
```suggestion
It maybe helpful to install the '[certificate_authority](https://github.com/cchandler/certificate_authority)' gem.
```

File: examples/generate_all_certs (hunk 2 above)
Comment (MSP-Greg) [reply examples/generate_all_certs:1]: I think I fixed this.  JFYI, my normal 'working' system is Windows (I started on DOS), and I forgot about the exe bit issue.

File: examples/generate_all_certs.md (hunk 3 above)
Comment (MSP-Greg) [reply examples/generate_all_certs.md]: Added the link

File: examples/generate_all_certs (hunk 1 above)
Comment (MSP-Greg) [reply examples/generate_all_certs:19]: I don't know, I thought the code should work if one cleared/removed all the certs, hence no error...

**Comment by MSP-Greg** [comment]
I just ran the code on macOS, and there are somethings I need to change.

**Comment by MSP-Greg** [comment]
I just pushed a commit that fixes the macOS issues.

Also, I updated the certs using my mac in another branch, the Tests CI passed, [results](https://github.com/MSP-Greg/puma/actions/runs/10354560901)

So, my testing shows this works with both Ubuntu & macOS.

File: examples/generate_all_certs (hunk 1 above)
Comment (dentarg) [reply examples/generate_all_certs:24]: It's a suggestion to use more simple and concise code :)

Comment (MSP-Greg) [reply examples/generate_all_certs:24]: Thanks, I missed that.  Still a bit distracted…

Comment (MSP-Greg) [reply examples/generate_all_certs]: Fixed...

**Comment by MSP-Greg** [comment]
Should I/we split this into two PR's?  One PR (this one) that just updates the 'generation' code, and another new PR that updates the ssl files?

When I go to the web page showing all the changed files, it's rather large.  Or, I'd rather see only the 'generation' code in this PR, and the ssl file updates in another?

**Review by dentarg — APPROVED** [review]

**Comment by dentarg** [comment]
We can rebase merge this, commits make sense to me

**Comment by MSP-Greg** [comment]
@dentarg

Thanks.  I was about to split it, and noticed it was closed.  Nice to have CI passing, except for the intermittent issues...

**Pull request merged**